refactoringminer/src/main/java/gr/uom/java/xmi/UMLModelASTReader.java

```diff
@@ -0,0 +1,564 @@
+        String packageName = getPackageName(file);
+        List<String> importedTypes = getImports(file);
+
+        PsiElement[] topLevelTypeDeclarations = file.getChildren();
+        for (PsiElement element : topLevelTypeDeclarations) {
+            if (element instanceof PsiClass) {
+                PsiClass topLevelClass = (PsiClass) element;
+                if (topLevelClass.isEnum()) {
+                    processEnumDeclaration(file, topLevelClass, packageName, sourceFilePath, importedTypes, comments);
+                } else {
+                    processTypeDeclaration(file, topLevelClass, packageName, sourceFilePath, importedTypes, comments);
+                }
+            }
+        }
+    }
+
+    @NotNull
+    private List<String> getImports(PsiFile file) {
+        PsiImportList imports = PsiUtils.findFirstChildOfType(file, PsiImportList.class);
+        return Arrays.stream(imports.getAllImportStatements())
+            .map(imp -> imp.isOnDemand()
+                ? imp.getImportReference().getText() + ".*"
+                : imp.getImportReference().getText())
+            .collect(Collectors.toList());
+    }
+
+    @NotNull
+    private String getPackageName(@NotNull PsiFile file) {
+        PsiPackageStatement packageDeclaration = PsiTreeUtil.findChildOfType(file, PsiPackageStatement.class);
+        String packageName;
+        if (packageDeclaration != null) {
+            packageName = packageDeclaration.getPackageName();
+        } else {
+            packageName = "";
+        }
+        return packageName;
+    }
+
+    /**
+     * Parse all comments in file
+     */
+    @NotNull
+    private List<UMLComment> extractInternalComments(@NotNull PsiFile file,
+                                                     @NotNull String sourceFile) {
+        Collection<PsiComment> psiComments = PsiTreeUtil.findChildrenOfType(file, PsiComment.class);
+        List<UMLComment> umlComments = new ArrayList<>();
+        for (PsiComment comment : psiComments) {
+            LocationInfo locationInfo = null;
+            if (comment.getTokenType() == JavaTokenType.END_OF_LINE_COMMENT) {
+                locationInfo = new LocationInfo(file, sourceFile, comment, LocationInfo.CodeElementType.LINE_COMMENT);
+            } else if (comment.getTokenType() == JavaTokenType.C_STYLE_COMMENT) {
+                locationInfo = new LocationInfo(file, sourceFile, comment, LocationInfo.CodeElementType.BLOCK_COMMENT);
+            }
+            if (locationInfo != null) {
+                String text = Formatter.format(comment);
+                UMLComment umlComment = new UMLComment(text, locationInfo);
+                umlComments.add(umlComment);
+            }
+        }
+        return umlComments;
+    }
+
+    private void distributeComments(List<UMLComment> compilationUnitComments, LocationInfo codeElementLocationInfo, List<UMLComment> codeElementComments) {
```

Comment:
In terms of Eclipse JDT a compilation unit means a file, right? let's rename it everywhere within this class